dashboard/app/helpers/levels_helper.rb

```diff
@@ -808,14 +809,24 @@ def include_multi_answers?(standalone)
   #
   # @param level_image [String] A base64-encoded image.
   # @param level_source_id [Integer, nil] The id of a LevelSource or nil.
+  # @param upgrade [Boolean] Whether to replace the saved image if level_image
+  #   is higher resolution
   # @returns [LevelSourceImage] A level source image, or nil if one was not
   # created or found.
-  def find_or_create_level_source_image(level_image, level_source_id)
+  def find_or_create_level_source_image(level_image, level_source_id, upgrade=false)
     level_source_image = nil
-    # Store the image only if the image is set, and the image has not been saved
+    # Store the image only if the image is set, and either the image has not been
+    # saved or the saved image is smaller than the provided image
     if level_image && level_source_id
       level_source_image = LevelSourceImage.find_by(level_source_id: level_source_id)
-      unless level_source_image
+      upgradable = false
+      if upgrade && level_source_image
+        old_image_size = ImageSize.path(level_source_image.s3_url)
+        new_image_size = ImageSize.new(Base64.decode64(level_image))
+        upgradable = new_image_size.width > old_image_size.width &&
+          new_image_size.height > old_image_size.height
+      end
```

Comment:
how often do we expect this case to get hit? It looks like we're doing an S3 request every time it does

Reply:
The `upgrade` argument is only true when editing solution blocks, so not often.